pkg/querier/worker/scheduler_processor.go

```diff
@@ -153,8 +153,14 @@ func (sp *schedulerProcessor) querierLoop(c schedulerpb.SchedulerForQuerier_Quer
 		go func() {
 			defer inflightQuery.Store(false)
 
+			// Create a per-request context and cancel it once we're done processing the request.
```

Comment:
Which problem do we want to solve here?

Reply:
The problem I was trying to solve here is the risk of any `SeriesChunksStreamReader` being leaked while it is running `SeriesChunksStreamReader.StartBuffering()`. By ensuring there's a per-request context, we can use that to cancel buffering when the request finishes if buffering doesn't terminate normally (which will only happen if all chunks are successfully read).

Reply:
Ok. I would just just update the comment to explain the use case here.

Reply:
Done in cc97b6d